lib/arjdbc/postgresql/oid_types.rb

```diff
@@ -40,6 +40,11 @@ def lookup_cast_type(sql_type)
       end
 
       def get_oid_type(oid, fmod, column_name, sql_type = '') # :nodoc:
+        # This can be called before the type map is created
```

Comment:
this is a good reason to have that one piece native (believe it was previously)
... and than hopefully no hacks shall be needed

Reply:
Do you mean "native" as in native AR or "native" as in Java based? This comment just made me realize that maybe it would be better to have this act more like AR in that the PG gem always returns a `PGresult` object and then its converted to an `AR::Result` object in `#exec_query` instead of having the java side create the `AR::Result` object directly

Reply:
@rdubya In general we do not want to double box data if we can avoid it.  Directly generating proper AR::Result seems best so long as we have enough knowledge to do that.

Reply:
I'll see what it turns into, this PR doesn't change how the result is generated so this should be ok.